multicluster/controllers/multicluster/common/helper.go

```diff
@@ -16,12 +16,23 @@ package common
 import (
 	"crypto/sha1" // #nosec G505: not used for security purposes
 	"encoding/hex"
+	"time"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 const labelIdentityHashLength = 16
 
+// CleanUpRetry is the retry when the clean up method
+// failed to clean up all stale resources.
+var CleanUpRetry = wait.Backoff{
+	Steps:    15,
+	Duration: 500 * time.Millisecond,
+	Factor:   2.0,
```

Comment:
The maximum wait duration could become 0.5s*2^15 = 4h+, is it expected?

Reply:
Yeah, it's too long, let me change it considering 4hrs might be meaningless to retry that long.

Reply:
Changed to 12.